packages/terra-form-select/src/native-select/NativeSelect.jsx

```diff
@@ -0,0 +1,224 @@
+     */
+    display: PropTypes.string,
+    /**
+     * Optional override of the default value.
+     */
+    value: PropTypes.string,
+  }),
+  /**
+   * Whether the field is required.
+   */
+  required: PropTypes.bool,
+  /**
+   * The value of the select. Can be a string or number.
+   */
+  value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+};
+
+const defaultProps = {
+  disabled: false,
+  isIncomplete: false,
+  isInvalid: false,
+  options: [],
+  required: false,
+};
+
+const createPlaceholder = (placeholder, intl) => {
+  if (!placeholder) {
+    return undefined;
+  }
+  const display = placeholder.display ? placeholder.display : intl.formatMessage({ id: 'Terra.form.select.defaultDisplay' });
+  const value = placeholder.value ? placeholder.value : defaultPlaceholderValue;
+  const attrs = placeholder.allowClear ? {} : { disabled: true, hidden: true };
+  return <option value={value} {...attrs}>{display}</option>;
+};
+
+const createOptions = options => options.map(current => {
+  if (current.childOptions) {
+    return (
+      <optgroup key={`${current.display} ${current.childOptions.length}`} label={current.display}>
+        {createOptions(current.childOptions)}
+      </optgroup>
+    );
+  }
+  return <option key={`${current.value}`} value={current.value}>{current.display}</option>;
+});
+
+const NativeSelect = ({
+  ariaDescribedBy,
+  ariaLabel,
+  attrs,
+  disabled,
+  defaultValue,
+  id,
+  intl,
+  isInvalid,
+  isIncomplete,
+  onChange,
+  options,
+  placeholder,
+  required,
+  value,
+  ...customProps
+}) => {
+  const [currentValue, setCurrentValue] = useState(defaultValue || getFirstValue(options, placeholder));
+  const refSelect = useRef();
+  const theme = React.useContext(ThemeContext);
+
+  const handleOnMouseDown = () => {
+    refSelect.current.setAttribute('data-focus-interaction', 'mouse');
+  };
+  const handleOnBlur = () => {
+    refSelect.current.setAttribute('data-focus-interaction', 'none');
+  };
+  const handleOnFocus = () => {
+    if (refSelect.current.getAttribute('data-focus-interaction') !== 'mouse') {
+      refSelect.current.setAttribute('data-focus-interaction', 'keyboard');
+    }
+  };
+  const handleOnChange = event => {
+    if (onChange) {
+      onChange(event);
+    }
+    if (!value) {
+      setCurrentValue(event.currentTarget.value);
+    }
+  };
+
+  const selectAttrs = {
+    'aria-describedby': ariaDescribedBy,
+    'aria-label': ariaLabel,
+    id,
+    disabled,
+    'aria-invalid': isInvalid || undefined,
+    required,
+    value: value || currentValue,
+  };
+
+  const nativeClassNames = cx(
+    'native',
+    theme.className,
+    { disabled },
+    { invalid: isInvalid },
+    { incomplete: required && isIncomplete },
+    { placeholder: isCurrentPlaceholder(selectAttrs.value, placeholder) },
+  );
+
+  return (
+    <div
+      {...customProps}
+      className={customProps.className ? `${nativeClassNames} ${customProps.className}` : nativeClassNames}
```

Comment:
@emilyrohrbough and I have been doing custom class names like this recently: https://github.com/cerner/terra-core/blob/master/packages/terra-card/src/Card.jsx#L45-L52
I think it's the cleanest pattern I've seen, but the bind thing is still kinda weird.

Reply:
Updated.